code/datums/components/z_parallax.dm

```diff
@@ -0,0 +1,60 @@
+/**
+ * Component that hooks into the client, listens for COMSIG_MOVABLE_Z_CHANGED, and depending on whether or not the
+ * Z-level has ZTRAIT_NOPARALLAX enabled, disable or reenable parallax.
+ */
+
+/datum/component/zparallax
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	var/client/tracked
+	var/mob/client_mob
+
+/datum/component/zparallax/Initialize(client/tracked)
+	. = ..()
+	if(!istype(tracked))
+		stack_trace("Component zparallax has been initialized outside of a client. Deleting.")
+		return COMPONENT_INCOMPATIBLE
+
+	src.tracked = tracked
+	client_mob = tracked.mob
+
+/datum/component/zparallax/RegisterWithParent()
+	RegisterSignal(client_mob, COMSIG_MOB_LOGOUT, .proc/mob_change)
+	RegisterSignal(client_mob, COMSIG_MOVABLE_Z_CHANGED, .proc/ztrait_checks)
+	RegisterSignal(client_mob, COMSIG_MOB_LOGIN, .proc/refresh_client)
+
+/datum/component/zparallax/UnregisterFromParent()
+	unregister_signals()
+
+	tracked = null
+	client_mob = null
```

Comment:
We use the RegisterWithParent and UnregisterFromParent procs to allow components to nicely transfer between parents, but if you're not doing any logic that involves the parent, then logic like this would just lead to duplicate registration errors if we ever did try and transfer parents